tools/testsys/src/crds.rs

```diff
@@ -159,13 +215,110 @@ impl<'a> CrdInput<'a> {
             "variant".to_string() => self.variant.to_string(),
             "kube-arch".to_string() => self.kube_arch(),
             "kube-variant".to_string() => self.kube_variant(),
+            "flavor".to_string() => some_or_null(&self.variant.variant_flavor().map(str::to_string)),
+            "version".to_string() => some_or_null(&self.variant.version().map(str::to_string)),
             "cluster-name".to_string() => cluster_name.to_string(),
             "instance-type".to_string() => some_or_null(&self.config.instance_type),
             "agent-role".to_string() => some_or_null(&self.config.agent_role),
             "conformance-image".to_string() => some_or_null(&self.config.conformance_image),
             "conformance-registry".to_string() => some_or_null(&self.config.conformance_registry),
         }
     }
+
+    /// Find the crd template file for the given test type
+    fn custom_crd_template_file_path(&self) -> Option<PathBuf> {
+        let test_type = &self.test_type.to_string();
+        // List all acceptable paths to the custom crd to allow users some freedom in the way
+        // `tests` is organized.
+        let acceptable_paths = vec![
+            // Check for <TEST-TYPE>.yaml in the top level directory
+            PathBuf::new().join(test_type).with_extension("yaml"),
+            // Check for <TESTSYS_FOLDER>/<TEST-TYPE>/<TEST-TYPE>.yaml
+            self.tests_directory
+                .join(test_type)
+                .join(test_type)
+                .with_extension("yaml"),
+            // Check for <TESTSYS_FOLDER>/<TEST-TYPE>/test.yaml
+            self.tests_directory.join(test_type).join("test.yaml"),
+            // Check for <TESTSYS_FOLDER>/<TEST-TYPE>/crd.yaml
+            self.tests_directory.join(test_type).join("crd.yaml"),
+            // Check for <TESTSYS_FOLDER>/<TEST-TYPE>/crds.yaml
+            self.tests_directory.join(test_type).join("crds.yaml"),
+            // Check for <TESTSYS_FOLDER>/shared/<TEST-TYPE>.yaml
```

Comment:
Do we need to support all three filenames? I'd prefer to be a bit more opinionated - `crd.yaml` only. `test.yaml` is a bit likely to be confused with `Test.toml`.